node/src/runner.rs

```diff
@@ -37,36 +40,58 @@ use crate::{
     config::NodeConfig,
     options::{Command, Options, RunOptions},
     regtest_options::ChainConfigOptions,
+    storage_config::StorageBackend,
 };
 
+/// Set up storage and initialize chainstate
+fn make_chainstate<B: 'static + storage::Backend>(
+    storage_backend: B,
+    chain_config: &Arc<ChainConfig>,
+    chainstate_config: chainstate::ChainstateConfig,
+) -> Result<Box<dyn ChainstateInterface>> {
+    let storage = chainstate_storage::Store::new(storage_backend)?;
+    let chainstate = chainstate::make_chainstate(
+        Arc::clone(chain_config),
+        chainstate_config,
+        storage,
+        DefaultTransactionVerificationStrategy::new(),
+        None,
+        Default::default(),
+    )?;
+    Ok(chainstate)
+}
+
 /// Initialize the node, giving caller the opportunity to add more subsystems before start.
 pub async fn initialize(
     chain_config: ChainConfig,
     node_config: NodeConfig,
 ) -> Result<subsystem::Manager> {
     let chain_config = Arc::new(chain_config);
 
-    // Initialize storage.
-    // TODO: Uses in-memory for now, will be configurable once multiple backends are implemented.
-    let storage = chainstate_storage::inmemory::Store::new_empty()?;
-
     // INITIALIZE SUBSYSTEMS
 
     let mut manager = subsystem::Manager::new("mintlayer");
     manager.install_signal_handlers();
 
+    // Pick default storage backend based on chain type
+    let backend_type =
+        node_config.storage.backend.unwrap_or_else(|| match chain_config.chain_type() {
+            ChainType::Signet | ChainType::Mainnet | ChainType::Testnet => StorageBackend::Lmdb,
+            ChainType::Regtest => StorageBackend::InMemory,
```

Comment:
Actually Regtest should be lmdb by default, as well. The only difference between Regtest and Mainnet is ChainConfig.

In-memory is only a choice to be made when necessary in Regtest.